lib/logstash/util/accessors.rb

```diff
@@ -48,44 +47,37 @@ def strict_set(accessor, value)
 
     def del(accessor)
       target, key = lookup(accessor)
-      unless target.nil?
-        target.is_a?(Array) ? target.delete_at(key.to_i) : target.delete(key)
-      end
+      return nil unless target
+      target.is_a?(Array) ? target.delete_at(key.to_i) : target.delete(key)
     end
 
     def include?(accessor)
-      target, key = lookup_path(accessor)
+      target, key = lookup(accessor)
```

Comment:
Yes, it makes sense once you understand the `a, b, ... = x, y, ...` semantic:
```ruby
irb(main):018:0> a, b = 1, 2
=> [1, 2]
irb(main):019:0> a
=> 1
irb(main):020:0> b
=> 2
```

`a, b = ...` is actually assigning an `Array` result for the RHS, first item to `a`, second item to `b` and so on. So what it on the RHS is automatically converted to an array. Now:

```ruby
irb(main):002:0> a, b = nil
=> [nil]
```
Since the RHS is a `[nil]` `Array` then 

```ruby
rb(main):021:0> [nil][0]
=> nil
irb(main):022:0> [nil][1]
=> nil
```